### PR TITLE
[175] Improve state transition diagram

### PR DESCRIPTION
**Pull request by arnauddieumegard**
- Drop State inside StateDef from diagram
- Set Core features to display on Core properties tab for StateUsage and StateDefinition
- Parse transition label to set Trigger and Effect referencing Actions.
- Add imports when needed

Bug: https://github.com/eclipse-syson/syson/issues/175

### CODE REVIEW

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ReferenceUsageImpl.java [hunk 1]
```diff
@@ -69,5 +72,18 @@ public EList<TextualRepresentation> getTextualRepresentation() {
         List<TextualRepresentation> textualRepresentation = new ArrayList<>();
         return new EcoreEList.UnmodifiableEList<>(this, SysmlPackage.eINSTANCE.getElement_TextualRepresentation(), textualRepresentation.size(), textualRepresentation.toArray());
     }
+    
+    @Override
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ReferenceUsageImpl.java:76]: Please a @generated NOT in a comment for this method

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ReferenceUsageImpl.java [hunk 2]
```diff
@@ -69,5 +72,18 @@ public EList<TextualRepresentation> getTextualRepresentation() {
...
+    public String getDeclaredName() {
+        Element foundActionUsage = this.getOwnedFeatureMembership().stream()
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ReferenceUsageImpl.java:78]: I'm not sure about that one. The ReferenceUsage is widely used in sysmlv2, for a large set of usages, not only ActionUsages.
You code should work for other kind of Usages too.

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ReferenceUsageImpl.java [hunk 3]
```diff
@@ -69,5 +72,18 @@ public EList<TextualRepresentation> getTextualRepresentation() {
...
+                .flatMap(fm -> fm.getTarget().stream())
+                .filter(elem -> elem instanceof ActionUsage)
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/ReferenceUsageImpl.java:80]: `.filter(elem -> elem instanceof ActionUsage)` -> `.filter(ActionUsage.class::isInstance)`

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/AbstractDefinitionNodeDescriptionProvider.java [hunk 4]
```diff
@@ -166,6 +167,7 @@ private NodePalette createNodePalette(NodeDescription nodeDescription, IViewDiag
                 .deleteTool(deleteTool.build())
                 .labelEditTool(editTool.build())
                 .edgeTools(edgeTools.toArray(EdgeTool[]::new))
+                .dropNodeTool(this.createDropFromDiagramTool(cache))
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/nodes/AbstractDefinitionNodeDescriptionProvider.java:170]: are you sure about adding that new behavior to all node descriptions in every existing diagram descriptions?

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/AcceptActionUsageImpl.java [hunk 5]
```diff
@@ -125,14 +127,37 @@ public Expression basicGetReceiverArgument() {
         // Ensure that you remove @generated or mark it @generated NOT
         return null;
     }
+    
+    /**
+     * Used to display the name of the trigger references in the {@code Related Elements} view.
+     */
+    @Override
+    public String getDeclaredName() {
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/AcceptActionUsageImpl.java:135]: The AcceptAction are still in review with this [PR](https://github.com/eclipse-syson/syson/pull/245) so you will not able to test it easily.

[the remark this answers is not in the document]
Comment (jerome-obeo) [reply backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/AcceptActionUsageImpl.java:135]: @AxelRICHARD The PR for AcceptActionUsage has not been pushed yet. So this is not yet available.
But the `getDeclaredName` code seems okay if the RefrenceUsage (the payload parameter) has a declared name.
This is not always the case.

[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/AcceptActionUsageImpl.java:135]: Ye that's what I add in the comment above ;)

**Pull request merged**